client.go

```diff
@@ -0,0 +1,69 @@
+package flagsmithapi
+
+import (
+	"fmt"
+
+	"github.com/go-resty/resty/v2"
+)
+
+const BaseAPIURL = "https://api.flagsmith.com/api/v1"
+
+type Client struct {
+	master_api_key string
+	baseURL        string
+	client         *resty.Client
+}
+
+func NewClient(masterAPIKey string, baseURL string) *Client {
+	if baseURL == "" {
+		baseURL = BaseAPIURL
+	}
+	c := &Client{master_api_key: masterAPIKey, baseURL: baseURL, client: resty.New()}
+	c.client.SetHeaders(map[string]string{
+		"Accept":        "application/json",
+		"Content-type":  "application/json",
+		"Authorization": "Api-Key " + c.master_api_key,
+	})
+	return c
+
+}
+
+// Get the feature state associated with the environment for a given feature
+func (c *Client) GetEnvironmentFeatureState(environmentAPIKey string, featureName string) (*FeatureState, error) {
+	url := fmt.Sprintf("%s/environments/%s/featurestates/", c.baseURL, environmentAPIKey)
+	fmt.Println("making request with", url)
+	result := struct {
+		Results []*FeatureState `json:"results"`
+	}{}
+	resp, err := c.client.R().
+		SetQueryParams(map[string]string{
+			"feature_name": featureName,
+		}).
+		SetResult(&result).
+		Get(url)
+
+	if err != nil {
+		return nil, err
+	}
+	if !resp.IsSuccess() {
+		return nil, fmt.Errorf("Error getting feature state: %s", resp.Status())
+
+	}
+	featureState := result.Results[0]
```

Comment:
What about if there aren't any results? Won't this throw some kind of unhandled index error?

Reply:
yep, my bad